src/lib/client-events.js

```diff
@@ -0,0 +1,10 @@
+const EventEmitter = require("events");
+const clientEvents = module.exports;
+
+const eventEmitter = new EventEmitter();
+
+clientEvents.on = eventEmitter.on.bind(eventEmitter);
+clientEvents.once = eventEmitter.once.bind(eventEmitter);
+clientEvents.emit = (eventName, eventPayload, ...args) => {
+  eventEmitter.emit(eventName, eventPayload, ...args);
+};
```

Comment:
I wonder if we can just do this:

```suggestion
module.exports = eventEmitter;
```